### PR TITLE
implements receiver and notification

### PR DESCRIPTION
**Pull request by humbertouchiyama**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -25,7 +27,20 @@
...
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.DEFAULT" />
+
+                <data
+                    android:scheme="tasky"
+                    android:host="agenda_item"
+                    android:pathPattern=".*" />
```
Comment (Kai-Cilliers) [inline app/src/main/AndroidManifest.xml:37]: AFAIK you don’t need to define the `pathPattern` here for a deeplink

File: app/src/main/java/com/humberto/tasky/core/alarm/presentation/AlarmReceiver.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        val id = intent.getStringExtra(ALARM_ID)
+        val title = intent.getStringExtra(ALARM_TITLE)?.takeIf { it.isNotEmpty() } ?: context.getString(R.string.no_title)
+        val description = intent.getStringExtra(ALARM_DESCRIPTION)
+        val itemType = intent.getStringExtra(ITEM_TYPE)
+        val selectedDate = intent.getLongExtra(ITEM_DATE, LocalDate.now().toEpochDay())
+
+        val activityIntent = Intent(context, MainActivity::class.java).apply {
+            data =
+                "tasky://agenda_item/${itemType}?agendaItemId=${id}&isEditing=${0}&selectedDateEpochDay=${selectedDate}"
+            .toUri()
```
Comment (Kai-Cilliers) [inline app/src/main/java/com/humberto/tasky/core/alarm/presentation/AlarmReceiver.kt:35]: It shouldn’t be needed to define the names like this. The follow will work:
```
"tasky://agenda_detail/${data.id}/${false}".toUri()
```

for a destination like this:

```
composable<AgendaDetailRoute>(
  deepLinks = listOf(
    navDeepLink<AgendaDetailRoute>(
      basePath = "tasky://agenda_detail”,
    )
  )
)

@Serializable
data class AgendaDetailRoute(
  val id: String,
  val isEditMode: Boolean,
)
```

Comment (humbertouchiyama) [reply app/src/main/java/com/humberto/tasky/core/alarm/presentation/AlarmReceiver.kt:35]: But the only non nullable val I have on my AgendaDetails route is the itemType, that's why I pass them like query params.

Comment (Kai-Cilliers) [reply app/src/main/java/com/humberto/tasky/core/alarm/presentation/AlarmReceiver.kt:35]: This table summarises the URI requirements for a Route nicely 😊

|                      | Path parameter                 | Query parameter                                                     |
| -------------------- | ------------------------------ | ------------------------------------------------------------------- |
| **Used for**             | required arguments             | optional arguments<br>collections (arrays and lists) | 
| **Example fields**       | `val id: String`<br>`val code: Int?`  | `val color: String? = null`<br>`val variants: List<String> = emptyList()` |
| **Generated URL format** | `/{id}/{code} `                  | `?color={color}&variants={variant1}&variants={variant2}`|
| **Example URL**          | `product/ABC/123`               | `product?color=red&variants=small&variants=medium` |

Comment (Kai-Cilliers) [reply app/src/main/java/com/humberto/tasky/core/alarm/presentation/AlarmReceiver.kt:35]: Also, this URI that I shared was for this model

```
@Serializable
data class AgendaDetailRoute(
  val id: String,
  val isEditMode: Boolean,
)
```

So if yours is different, then the URI will also be.

**Pull request merged**